ContosoPizza/Features/Users/Delete/DeleteUserHandler.cs

```diff
@@ -3,24 +3,39 @@
 using Microsoft.EntityFrameworkCore;
 using Nudes.Retornator.AspnetCore.Errors;
 using Nudes.Retornator.Core;
+using System.Security.Claims;
 
 namespace ContosoPizza.Features.Users.Delete
 {
     public class DeleteUserHandler : IRequestHandler<DeleteUserRequest, Result>
     {
         private readonly ContosoPizzaContext db;
+        private readonly IHttpContextAccessor httpContextAccessor;
 
-        public DeleteUserHandler(ContosoPizzaContext db)
+
+        public DeleteUserHandler(ContosoPizzaContext db, IHttpContextAccessor httpContextAccessor)
         {
             this.db = db;
+            this.httpContextAccessor = httpContextAccessor;
         }
         public async Task<Result> Handle(DeleteUserRequest request, CancellationToken cancellationToken)
         {
-            var user = await db.Users.FirstOrDefaultAsync(p => p.Id == request.Id, cancellationToken);
-            if (user is null)
+            var userContext = httpContextAccessor.HttpContext.User;
+            var userClaims = userContext.Claims
+                .Where(claim => claim.Type == "Claim")
+                .Select(d => d.ToString())
```

Comment:
Selecionar o `Enum.Parse<EnumType>(d.Value)` aqui no lugar do ToString(), to list é desnecessário